File: Assets/ProjectGame/Lib/data/enum/GameObject_Character_Prefab/GameObject_Character_PrefabID.py
from enum import IntEnum
from typing import Callable, List, Any


class GameObject_Character_PrefabID(IntEnum):
    NONE = 0  # デフォルト値（C#のNoneに相当）
    Ange = 1  # アンジュ
    Ryushen = 2  # 緑仙
    Belmond = 3  # ベルモンド
    Himawari = 4  # 本間ひわまり
    Mashiro = 5  # ましろ
    Kuzuha = 6  # 葛葉
    MAX = 7

def to_int(id: "GameObject_Character_PrefabID") -> int:
    return int(id)


def to_GameObject_Character_PrefabID(id: int) -> "GameObject_Character_PrefabID":
    return GameObject_Character_PrefabID(id)


def to_index(id: "GameObject_Character_PrefabID") -> int:
    return int(id) - 1


def for_id(action: Callable[["GameObject_Character_PrefabID"], None]):
    if action is None:
        raise ValueError("action cannot be None")
    start = GameObject_Character_PrefabID.Ange.value
    for i in range(start, GameObject_Character_PrefabID.MAX.value):
        try:
            value = GameObject_Character_PrefabID(i)
            action(value)
        except ValueError:
            continue  # 未定義の値はスキップ


def find_all(predicate: Callable[["GameObject_Character_PrefabID"], bool]) -> List["GameObject_Character_PrefabID"]:
    if predicate is None:
        raise ValueError("predicate cannot be None")
    results: List["GameObject_Character_PrefabID"] = []
    start = GameObject_Character_PrefabID.Ange.value
    for i in range(start, GameObject_Character_PrefabID.MAX.value):
        try:
            value = GameObject_Character_PrefabID(i)
            if predicate(value):
                results.append(value)
        except ValueError:
            continue
    return results


def find(predicate: Callable[["GameObject_Character_PrefabID"], bool]) -> "GameObject_Character_PrefabID":
    if predicate is None:
        raise ValueError("predicate cannot be None")
    start = GameObject_Character_PrefabID.Ange.value
    for i in range(start, GameObject_Character_PrefabID.MAX.value):
        try:
            value = GameObject_Character_PrefabID(i)
            if predicate(value):
                return value
        except ValueError:
            continue
    return GameObject_Character_PrefabID.NONE

File: Assets/ProjectGame/Lib/data/enum/GameObject_Character_Prefab/GameObject_Character_PrefabID.js
// GameObject_Character_PrefabID - Pure JavaScript Version
export const GameObject_Character_PrefabID = {
    None: 0,  // デフォルト値（C#互換）
    Ange: 1,  // アンジュ
    Ryushen: 2,  // 緑仙
    Belmond: 3,  // ベルモンド
    Himawari: 4,  // 本間ひわまり
    Mashiro: 5,  // ましろ
    Kuzuha: 6,  // 葛葉
    Max: 7
};

export const GameObject_Character_PrefabIDExtensions = {
    /**
     * Enumを数値に変換
     */
    toInt(id) {
        return Number(id);
    },

    /**
     * 数値をEnumに変換
     */
    toGameObject_Character_PrefabID(id) {
        return id;
    },

    /**
     * 0-based indexに変換
     */
    toIndex(id) {
        return Number(id) - 1;
    },

    /**
     * すべてのIDに対して処理を実行
     */
    forID(action) {
        if (typeof action !== 'function') {
            throw new Error('action must be a function');
        }
        const start = GameObject_Character_PrefabID.Ange;
        const max = GameObject_Character_PrefabID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_Character_PrefabID).includes(id)) {
                action(id);
            }
        }
    },

    /**
     * 条件に合うすべてのIDを返す
     */
    findAll(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const results = [];
        const start = GameObject_Character_PrefabID.Ange;
        const max = GameObject_Character_PrefabID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_Character_PrefabID).includes(id)) {
                if (predicate(id)) {
                    results.push(id);
                }
            }
        }
        return results;
    },

    /**
     * 条件に合う最初のIDを返す（見つからなければ None）
     */
    find(predicate) {
        if (typeof predicate !== 'function') {
            throw new Error('predicate must be a function');
        }
        const start = GameObject_Character_PrefabID.Ange;
        const max = GameObject_Character_PrefabID.Max;
        for (let id = start; id < max; id++) {
            if (Object.values(GameObject_Character_PrefabID).includes(id)) {
                if (predicate(id)) {
                    return id;
                }
            }
        }
        return GameObject_Character_PrefabID.None;
    }
};
